src/ccache.c

```diff
@@ -1371,8 +1371,10 @@ update_manifest_file(void)
 		old_size = file_size(&st);
 	}
 
+	bool save_timestamp = (conf->sloppiness & SLOPPY_FILE_STAT_MATCHES) || output_is_precompiled_header;
```

Comment:
I understand that the `|| output_is_precompiled_header` part is because of the CI error you got, but did you investigate why the test failed?

Reply:
The CI error is important and helpful, however I also found the answer in the source code.

https://github.com/ccache/ccache/blob/1a4929bfd73e63ebd1b369ee116f9c57f1931824/src/manifest.c#L391-L399

ccache will check mtime in file_info if it detected the cmd is about pch. So, I keep the timestamp if pch is detected.

Reply:
Excellent, thanks.